docs/developer_guide/e2e_test.md

```diff
@@ -94,4 +94,36 @@ ginkgo run -r -v --skip-file "admission_controller_test.go" -- -yk-namespace "yu
 * Delete the cluster after we finish testing (this step is optional).
 ```shell
 ./scripts/run-e2e-tests.sh -a cleanup -n yktest
-```
\ No newline at end of file
+```
+
+## Checking E2E Test Logs
+
+The logs of failed E2E tests are stored locally and uploaded to the GitHub Action Artifact. This section explains how to retrieve and understand these logs.
+
+### Local E2E Test Logs
+
+When an E2E test fails, logs are saved locally in the `yunikorn-k8shim/build/e2e/{suite}/` directory. The logs are categorized as follows:
+
+- `{specName}_k8sClusterInfo.txt`: Contains information about the Kubernetes cluster state at the time of the test.
+- `{specName}_ykContainerLog.txt`: Includes logs from the YuniKorn containers.
+- `{specName}_ykFullStateDump.json`: Provides a full state dump of YuniKorn.
+
+### Steps to Retrieve Local Logs
+
+1. Run the E2E test using `make e2e_test`.
```

Comment:
Can we recommend to use `./scripts/run-e2e-tests.sh -a "test" -n "yk8s"` instead of `make e2e_test`? `run-e2e-tests.sh` is easy to use. Also it is the command used by CI.